### PR TITLE
Implemented fix to exclude specified file path using `!` symbol

### LINKED ISSUE #1083: Doesn't work the `excludeFiles` field with the bang(`!`)
When attempting to negate relative paths using the bang (`!`) symbol in the `excludeFiles` field, path resolution appears to fail. When `!./path/to` is provided, it is treated as `/abs-path/!path/to`, which isn't the desired behavior.

### PR DESCRIPTION
**Pull request by YusukeHirao**
Fixes #1083

## What is the purpose of this PR?

- [x] Fix bug
- [ ] Fix typo
- [ ] Update specs
- [ ] Add new rule
- [ ] Add new parser
- [ ] Improve or refactor rules
- [ ] Add a new core feature
- [ ] Improve or refactor core features
- [ ] Update documents
- [ ] Others

## Checklist

Fill out the checks for the applicable purpose.

- [x] Fix bug
  - [x] Add the test that can check whether resolved the issue.
- [ ] Update specs
  - [ ] Element
    - [ ] Update **[global attributes](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/html-spec/src/spec-common.attributes.json)**
    - [ ] Update **[content models](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/html-spec/src/spec-common.contents.json)**
    - [ ] Add also ARIA that you explored **ARIA in HTML [Recommendation](https://www.w3.org/TR/html-aria/) and [Editor's draft](https://w3c.github.io/html-aria/)**
  - [ ] Attribute
    - [ ] Update **[global attributes](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/html-spec/src/spec-common.attributes.json)**
    - [ ] Update **[IDL attribute](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/parser-utils/src/idl-attributes.ts)**
- [ ] Add new rule
  - [ ] Add a test
  - [ ] Add a README document
    - [ ] Create an issue that requests translating the doc if you want.
  - [ ] Add to [index](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/rules/src/index.ts)
  - [ ] Add JSON schema to [schema.json](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/rules/schema.json)
  - [ ] Add to [`@markuplint/config-presets`](https://github.com/markuplint/markuplint/tree/main/packages/%40markuplint/config-presets)
    - [ ] And add to [Rulesets of base presets](https://markuplint.dev/docs/guides/presets#rulesets-of-base-presets)
      - [ ] Add to a [doc file](https://github.com/markuplint/markuplint/blob/main/website/docs/guides/presets.md)
      - [ ] Add to a [doc file (ja)](https://github.com/markuplint/markuplint/blob/main/website/i18n/ja/docusaurus-plugin-content-docs/current/guides/presets.md), or create an issue that requests translating the doc if you want.
  - [ ] Add to the [Playground](https://playground.markuplint.dev/)
    - [ ] Add to a [config object](https://github.com/markuplint/markuplint/blob/main/playground/src/ml-playground-home.svelte)
- [ ] Add new parser
  - [ ] Add a test
    - [ ] Do a test with some parser.
  - [ ] Add to [initializer](https://github.com/markuplint/markuplint/blob/main/packages/markuplint/src/cli/init/index.ts)

### CODE REVIEW

File: packages/@markuplint/file-resolver/src/ml-file/ml-file.ts [hunk 1]
```diff
@@ -73,6 +74,16 @@ export class MLFile {
 		return '';
 	}
 
+	ignored(globPath: string | readonly string[]) {
+		if (typeof globPath === 'string') {
```
Comment (kagankan) [inline packages/@markuplint/file-resolver/src/ml-file/ml-file.ts:78]: Should `minimatch` be used if `globPath` is not an array?
Intuitively, I prefer the following code

```ts
const normalizedPaths = (typeof globPath === 'string' ? [globPath] : globPath).map(p => pathNormalize(p, true));
```

File: packages/@markuplint/file-resolver/src/ml-file/ml-file.ts [hunk 2]
```diff
@@ -131,12 +142,26 @@ async function stat(filePath: string) {
 	}
 }
 
-function pathNormalize(filePath: string) {
+function pathNormalize(filePath: string, relative?: boolean) {
```
Comment (kagankan) [inline packages/@markuplint/file-resolver/src/ml-file/ml-file.ts:145]: Don't you need a default value?

```ts
function pathNormalize(filePath: string, relative = false) {
```

File: packages/markuplint/src/api/ml-engine.ts [hunk 3]
```diff
@@ -211,12 +211,10 @@ export default class MLEngine extends Emitter<MLEngineEventMap> {
...
 		// Exclude
 		const excludeFiles = configSet.config.excludeFiles ?? [];
-		for (const excludeFile of excludeFiles) {
-			if (this.#file.matches(excludeFile)) {
-				this.emit('exclude', this.#file.path, excludeFile);
-				fileLog('Excludes the file: %s', this.#file.path);
-				return null;
-			}
+		if (this.#file.ignored(excludeFiles)) {
+			// this.emit('exclude', this.#file.path, excludeFile);
```
Comment (kagankan) [inline packages/markuplint/src/api/ml-engine.ts:215]: Is this comment necessary?
Seemed like you could delete it if it is not needed.

File: website/i18n/ja/docusaurus-plugin-content-docs/current/configuration/properties.md [hunk 4]
```diff
@@ -251,11 +251,11 @@ interface Config {
 
 ### `excludeFiles`
 
-必要であれば、ファイルを除外できます。値は**設定ファイルからの相対パスか絶対パス**が必要です。パスはglob形式も可能です。([minimatch](https://www.npmjs.com/package/minimatch)を用いて解決されます)
+必要であれば、ファイルを除外できます。値は**設定ファイルからの相対パスか絶対パス**が必要です。パスはglob形式も可能です。否定を表す`!`シンボルを使うこともできます。後から指定したものが優先されます。パターンは`.gitignore`の[仕様](https://git-scm.com/docs/gitignore)に従って動作します。（[node-ignore](https://github.com/kaelzhang/node-ignore)を用いて解決されます）
```
Comment (kagankan) [inline website/i18n/ja/docusaurus-plugin-content-docs/current/configuration/properties.md:254]: Wouldn't this be a more appropriate scope for the link?

```md
[`.gitignore`の仕様](https://git-scm.com/docs/gitignore)
```

**Review by kagankan — APPROVED** [review]
I added some comments, but I'll leave it up to you to revise them or not. 🙏

File: website/i18n/ja/docusaurus-plugin-content-docs/current/configuration/properties.md (hunk 4 above)
Comment (YusukeHirao) [reply website/i18n/ja/docusaurus-plugin-content-docs/current/configuration/properties.md:254]: Right. Thanks!

File: packages/markuplint/src/api/ml-engine.ts (hunk 3 above)
Comment (YusukeHirao) [reply packages/markuplint/src/api/ml-engine.ts:215]: Thanks, I'll remove it.

File: packages/@markuplint/file-resolver/src/ml-file/ml-file.ts (hunk 2 above)
Comment (YusukeHirao) [reply packages/@markuplint/file-resolver/src/ml-file/ml-file.ts:145]: Right. It is a quirk of mine. I'll fix it.

**Pull request merged**